com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/cache/ResourceCacheKey.java

```diff
@@ -15,65 +15,26 @@
 package com.dynamo.bob.cache;
 
 import java.security.MessageDigest;
-import java.util.List;
-import java.util.ArrayList;
-import java.util.Map;
-import java.util.Set;
-import java.util.HashSet;
-import java.util.Collections;
+import java.security.NoSuchAlgorithmException;
 import java.io.IOException;
 import java.math.BigInteger;
 
-import com.dynamo.bob.Task;
 import com.dynamo.bob.fs.IResource;
-import com.dynamo.bob.archive.EngineVersion;
 
 public class ResourceCacheKey {
 
-	/*
-	 * A set of options which have an impact on the created resources
-	 * and must be included when calculating the resource key
-	 */
-	private static Set<String> options = new HashSet<String>();
-
-	/**
-	 * Add an option that should be included in the resource key
-	 * calculation
-	 */
-	public static void includeOption(String option) {
-		options.add(option);
-	}
-
-
 	/**
 	 * Calculate the key to use when caching a resource.
-	 * The key is created from the task used which created the resource, the
-	 * bytes of the resource itself, as well as engine sha and project options.
-	 * @param task The task which created the resource
-	 * @param projectOptions The project options
+	 * The key is created from the task used which created the resource and the
+	 * bytes of the resource path.
+	 * @param taskSignature The task which created the resource
 	 * @param resource The resource to calculate cache key for
 	 * @return The cache key as a hex string
 	 */
-	public static String calculate(Task task, Map<String, String> projectOptions, IResource resource) throws RuntimeException, IOException {
-		MessageDigest digest = task.calculateSignatureDigest();
-
+	public static String calculate(byte[] taskSignature, IResource resource) throws RuntimeException, IOException, NoSuchAlgorithmException {
+		MessageDigest digest = MessageDigest.getInstance("SHA1");
+        digest.update(taskSignature);
 		digest.update(resource.getPath().getBytes());
```

Comment:
I didn't get why it's needed, but I decided to keep it as is.
This function does not that much now.

Reply:
Hmm, yeah, it should be the protobuf file formats that are important, not the engine version?